documentation/asciidoc/computers/raspberry-pi/boot-http.adoc

```diff
@@ -0,0 +1,113 @@
+
+=== Configuration Properties
+
+==== NET_INSTALL_ENABLED
+
+By default net install is enabled on Pi4 and Pi400. On CM boards it is disabled by default. 
+
+When net install is enabled it detects a keyboard and displays the Net Install screen on boot. You can switch this behaviour on and off with this property.
+
+To enable network install add, `NET_INSTALL_ENABLED=1`, or to disable network install add `NET_INSTALL_ENABLED=0`.
+
+This setting is ignored and net install / keyboard detection on boot is disabled if `DISABLE_HDMI=1` is set.
+
+==== NET_INSTALL_KEYBOARD_WAIT
+
+If net install is enabled, the bootloader waits to detect a keyboard and the `SHIFT` key to initiate network install. You can change the length of this wait in milliseconds with this property,
+
+Default: `900`
+
+Adding `NET_INSTALL_KEYBOARD_WAIT=0` disables the keyboard wait, although network install can still be initiated if no boot files are found and USB boot `4` is in `BOOT_ORDER`.
+
+NOTE: Testing suggests Keyboard / shift detection takes at least 750ms.
+
+==== HTTP_HOST
+
+If a HTTP download is initiated boot.img and boot.sig are downloaded from this server.
+
+Default: `fw-download-alias1.raspberrypi.com`
+
+Invalid host names will be ignored.
+If `HTTP_HOST` is set then HTTPS is disabled and plain HTTP used instead.
+You should be able to specify an IP address and avoid the need for a DNS lookup.
+Don`t include the HTTP scheme or any forward slashes in the hostname.
+
+==== HTTP_PORT
+
+By default HTTP uses port `443` because HTTPS is enabled when using the Raspberry Pi host. If `HTTP_HOST` is changed then HTTPS is not supported and HTTP will be used instead, so port `80` becomes the default.
+
+You can override this and specify a different port, but this is ignored if `HTTP_HOST` is not set.
+
+==== HTTP_PATH
+
+You can change the path used to download files with this property.
+
+Default: `net_install`
+
+The case of the path *is* significant.
+Use forward (Linux) slashes for the path separator.
+Leading and trailing forward slashes are not required.
+Any value for `HTTP_PATH` will only be used if `HTTP_HOST` is also set.
+
+==== IMAGER_REPO_URL
+
+The imager application is configured with a json file which it downloads on start up.
+
+Default: http://downloads.raspberrypi.org/os_list_imagingutility_v3.json
+
+You can change the URL and json file used by the imager application to get it to supply your own images.
+Test this with the Windows or Linux Imager applications by passing the URL via the `--repo` argument.
+
+=== Keys
+
+All HTTP downloads must be signed. The Raspberry Pi public key will be used to verify the network install image unless `HTTP_HOST` is set and a custom public key is included in the eeprom.
```

Comment:
Might be worth putting the "and" in bold: "...unless `HTTP_HOST` is set **and** a custom public key..." as I initially misinterpreted this to mean that if `HTTP_HOST` is set then it won't use the Raspberry Pi public key.

Reply:
Now I'm even more confused - "This allows you to host the Raspberry Pi network install images on your own server." disagrees with "If they change the host then we assume we can no longer use our key." ??